database/consensus.go

```diff
@@ -184,3 +193,151 @@ func (db *Db) GetGenesis() (*types.Genesis, error) {
 	row := rows[0]
 	return types.NewGenesis(row.ChainID, row.Time, row.InitialHeight), nil
 }
+
+// CheckIfBlockIsMissing checks if block is already stored in database
+func (db *Db) CheckIfBlockIsMissing(height int64) bool {
+	var block []dbtypes.BlockRow
+	stmt := `SELECT * FROM block WHERE height = $1`
+
+	err := db.Sqlx.Select(&block, stmt, height)
+	if err != nil {
+		return true
+	}
+	if len(block) != 0 {
+		return false
+	}
+
+	return true
+}
+
+// UpdateBlocksInDatabase updates given block in database
+func (db *Db) UpdateBlockInDatabase(block *tmctypes.ResultBlock, blockResults *tmctypes.ResultBlockResults) error {
```

Comment:
Why do we need to re-create all these methods that are already part of the `Database` interface? Can't we simply use `db.SaveBlock`, `db.SaveTransaction` and `db.SaveMessage` instead?

Reply:
Sure, I updated the code!